File: app/src/main/java/fr/socket/florian/dhome/view/connections/ConnectionAdapter.java
package fr.socket.florian.dhome.view.connections;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import fr.socket.florian.dhome.R;
import fr.socket.florian.dhome.database.Connection;

class ConnectionAdapter extends RecyclerView.Adapter<ConnectionViewHolder> {

    private List<Connection> connections;

    ConnectionAdapter() {
        connections = new ArrayList<>();
    }

    void update(List<Connection> connections) {
        this.connections = connections;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ConnectionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.connection_row, parent, false);
        return new ConnectionViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ConnectionViewHolder connectionViewHolder, int i) {
        connectionViewHolder.update(connections.get(i));
    }

    @Override
    public int getItemCount() {
        return connections.size();
    }
}

File: app/src/main/java/fr/socket/florian/dhome/view/connections/ConnectionAdapter.kt
package fr.socket.florian.dhome.view.connections

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import fr.socket.florian.dhome.R
import fr.socket.florian.dhome.database.Connection
import java.util.*

internal class ConnectionAdapter : RecyclerView.Adapter<ConnectionViewHolder>() {

    private var connections: List<Connection> = ArrayList()

    fun update(connections: List<Connection>) {
        this.connections = connections
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, i: Int): ConnectionViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.connection_row, parent, false)
        return ConnectionViewHolder(view)
    }

    override fun onBindViewHolder(connectionViewHolder: ConnectionViewHolder, i: Int) {
        connectionViewHolder.update(connections[i])
    }

    override fun getItemCount(): Int {
        return connections.size
    }
}
